core/src/main/java/io/paradoxical/cassieq/modules/ResourceAllocationModule.java

```diff
@@ -41,7 +42,19 @@ public ResourceAllocatorStrategySelector(AllocationConfig config, Injector injec
         public ResourceAllocator.Factory getFactory() {
             switch (config.getStrategy()) {
                 case CLUSTER:
-                    return (config, setSupplier, claimSetConsumer) -> getChildInjector(config, setSupplier, claimSetConsumer).getInstance(HazelcastResourceAllocater.class);
+                    return (config, setSupplier, claimSetConsumer) -> {
+                        final Injector childInjector = getChildInjector(config, setSupplier, claimSetConsumer);
+
+                        // create a clustered allocator and wrap it with an eventable allocator
+                        final ResourceAllocator clusteredResourceAllocator = childInjector.getInstance(HazelcastResourceAllocater.class);
+
+                        return childInjector.createChildInjector(new AbstractModule() {
+                            @Override
+                            protected void configure() {
+                                bind(ResourceAllocator.class).toInstance(clusteredResourceAllocator);
+                            }
```

Comment:
if all the children are assisted or singletons seems like a Factory would work nicely?